### PR TITLE
[Monolog] Added documentation about ElasticsearchLogstashHandler

### LINKED ISSUE #12198: [Monolog] Added ElasticsearchLogstashHandler
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#32360
| PR author(s) | @lyrixx
| Merged in    | 4.4

### PR DESCRIPTION
**Pull request by lyrixx**
Fixes #12198

### CODE REVIEW

File: logging/handlers.rst [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+This handler dials directly with the HTTP interface of Elasticsearch. This means
+it will slow down your application if Elasticsearch takes times to answer. Even
+if all HTTP calls are done asynchronously.
+
+In a development environment, it's fine to keep the default configuration: for
+each log, an HTTP request will be made to push the log to Elasticsearch.
+
+In a production environment, it's highly recommended to wrap this handler in a
+handler with buffering capabilities (like the FingersCrossedHandler, or
+BufferHandler) in order to call Elasticsearch only once with a bulk push. For
```
Comment (OskarStark) [inline logging/handlers.rst]:
```suggestion
``BufferHandler``) in order to call Elasticsearch only once with a bulk push. For
```

btw. is it Buffer or Buffere**D** handler?

File: logging/handlers.rst [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        </container>
+
+    .. code-block:: php
+
+        // config/packages/prod/monolog.php
+        $container->loadFromExtension('monolog', [
+            'handlers' => [
+                'es' => [
+                    'type' => 'service',
+                    'id' => 'Symfony\\Bridge\\Monolog\\Handler\\ElasticsearchLogstashHandler',
```
Comment (OskarStark) [inline logging/handlers.rst]: same question as above with the use statement.

**Review by OskarStark — COMMENTED** [review]
Thank you Greg 👍 

I left some comments.

File: logging/handlers.rst [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+            </services>
+        </container>
+
+    .. code-block:: php
+
+        // config/services.php
+        use App\Logger\SessionRequestProcessor;
+        use Monolog\Formatter\LineFormatter;
+
+        $container->register('Symfony\\Bridge\\Monolog\\Handler\\ElasticsearchLogstashHandler');
```
[the remark this answers is not in the document]
Comment (lyrixx) [reply logging/handlers.rst]: Sure :+1:

File: logging/handlers.rst (hunk 2 above)
Comment (lyrixx) [reply logging/handlers.rst]: :+1:

**Comment by lyrixx** [comment]
@OskarStark Thanks for the review. I updated the PR

File: logging/handlers.rst [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+In a development environment, it's fine to keep the default configuration: for
+each log, an HTTP request will be made to push the log to Elasticsearch.
+
+In a production environment, it's highly recommended to wrap this handler in a
+handler with buffering capabilities (like the ``FingersCrossedHandler`` or
+``BufferHandler``) in order to call Elasticsearch only once with a bulk push. For
+even better performance and fault tolerance, a proper `ELK stack`_ is recommended.
+
+To use it, declare it as a service:
```
Comment (OskarStark) [inline logging/handlers.rst]:
```suggestion
You need to declare it as a service first:
```

File: logging/handlers.rst [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+
```
Comment (OskarStark) [inline logging/handlers.rst]:
```suggestion

```

File: logging/handlers.rst [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+    .. code-block:: php
+
+        // config/services.php
+        use App\Logger\SessionRequestProcessor;
+        use Monolog\Formatter\LineFormatter;
+        use Symfony\\Bridge\\Monolog\\Handler\\ElasticsearchLogstashHandler;
+
+        $container->register(ElasticsearchLogstashHandler::class);
+
+then reference it in monolog configuration:
```
Comment (OskarStark) [inline logging/handlers.rst]:
```suggestion
then use this referenced service in your monolog configuration:
```

File: logging/handlers.rst [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+
+
```
Comment (OskarStark) [inline logging/handlers.rst]:
```suggestion

```

File: logging/handlers.rst [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+                    type="service"
+                    id="Symfony\Bridge\Monolog\Handler\ElasticsearchLogstashHandler"
+                />
+            </monolog:config>
+        </container>
+
+    .. code-block:: php
+
+        use Symfony\\Bridge\\Monolog\\Handler\\ElasticsearchLogstashHandler;
+        // config/packages/prod/monolog.php
```
Comment (OskarStark) [inline logging/handlers.rst]:
```suggestion
        // config/packages/prod/monolog.php
       use Symfony\Bridge\Monolog\Handler\ElasticsearchLogstashHandler;        
```

File: logging/handlers.rst [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+                    name="es"
+                    type="service"
+                    id="Symfony\Bridge\Monolog\Handler\ElasticsearchLogstashHandler"
+                />
+            </monolog:config>
+        </container>
+
+    .. code-block:: php
+
+        use Symfony\\Bridge\\Monolog\\Handler\\ElasticsearchLogstashHandler;
```
Comment (OskarStark) [inline logging/handlers.rst]:
```suggestion

```

File: logging/handlers.rst [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+                <service id="Symfony\Bridge\Monolog\Handler\ElasticsearchLogstashHandler" />
+            </services>
+        </container>
+
+    .. code-block:: php
+
+        // config/services.php
+        use App\Logger\SessionRequestProcessor;
+        use Monolog\Formatter\LineFormatter;
+        use Symfony\\Bridge\\Monolog\\Handler\\ElasticsearchLogstashHandler;
```
Comment (OskarStark) [inline logging/handlers.rst]:
```suggestion
        use Symfony\Bridge\Monolog\Handler\ElasticsearchLogstashHandler;
```

**Review by OskarStark — COMMENTED** [review]
Second round 😄

**Comment by lyrixx** [comment]
Second "thanks" :)

**Comment by lyrixx** [comment]
Could be merge this one?

**Comment by OskarStark** [comment]
@javiereguiluz could you please give us a final review here? Thanks 🙏

File: logging/handlers.rst [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
+========
+
+ElasticsearchLogstashHandler
+----------------------------
+
+.. versionadded:: 4.4
+
+    The ``ElasticsearchLogstashHandler`` was introduced in Symfony 4.4.
+
+This handler dials directly with the HTTP interface of Elasticsearch. This means
```
Comment (javiereguiluz) [inline logging/handlers.rst]: This phrase looks strange to me. Maybe `dials with` should be `deals with`?

File: logging/handlers.rst [hunk 12]
```diff
@@ -0,0 +1,106 @@
...
+    .. code-block:: php
+
+        // config/services.php
+        use App\Logger\SessionRequestProcessor;
+        use Monolog\Formatter\LineFormatter;
+        use Symfony\Bridge\Monolog\Handler\ElasticsearchLogstashHandler;
+
+        $container->register(ElasticsearchLogstashHandler::class);
+
+then use this referenced service in your monolog configuration:
```
Comment (javiereguiluz) [inline logging/handlers.rst]: `then` -> `Then`

**Review by javiereguiluz — APPROVED** [review]
Only two minor comments. Nice work! Thanks.

File: logging/handlers.rst (hunk 11 above)
Comment (lyrixx) [reply logging/handlers.rst]: I don't know. I'm not fluent :) I meant PHP dialog with ES via the HTTP interface. (You can communicate with ES with the "java interface").

Anyway, I updated the sentence with your proposal

**Comment by OskarStark** [comment]
Please check the warnings by DOCtor-RST

**Review by OskarStark — APPROVED** [review]

**Comment by javiereguiluz** [comment]
Grégoire, thanks for providing these docs!

**Pull request merged**